test/src/main/java/derp/SomeEnum.java

```diff
@@ -0,0 +1,22 @@
+package derp;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+public enum SomeEnum {
+  SomeValue("some value");
+
+  public final String value;
+
+  SomeEnum(String value) {
+    this.value = value;
+  }
+
+  @JsonCreator
```

Comment:
To support these, we may want yet another post-gen mod on enums.  Probably not hard to add this one in.

Reply:
I thought that's how they were generated already since that's such a standard approach, but you're right they don't work that way.  I will create a ticket to support enums as generated by the RAML gen.